proposals/2161/registry.md

```diff
@@ -0,0 +1,280 @@
+
+### CI and Documentation
+
+#### CI validation
+- The registry is consistent:
+  - Validate custom facet prefixes match the registered name
+- The registry has the required fields
+- Linting.
+- Custom facet schemas are valid and validated against examples.
+
+#### The registry will be used to publish documentation in CI
+- A page similar to our ecosystem page that lists producers, consumers, links to their documentation and what facets they support.
+- Custom facets on their schema URL like the current core facets:
+ Ex: https://openlineage.io/spec/facets/1-0-1/ColumnLineageDatasetFacet.json
+- Generated doc from the json schemas of the facets. Publish them on openlineage.io
+
+#### Additional documentation
+We can create a documentation page per use case to document what facet can be use in what case:
+
+- Compliance with privacy laws (GDPR, CCPA, …)
+- Compliance with Banking regulation (BCBS-239)
+- Data reliability, data quality
+- Data discovery, data catalog
+- Data Governance
+- Data Lineage
+- …
+
+We should have a page explaining how a custom facet can be promoted to a core facet through the OpenLineage Proposal process.
+
+### The Registry
+
+We propose a self contained registry hosted in the OpenLineage repository.
+
+**TL;DR: The registry defines consumers and producers names and contains all their custom facets.**
+
+Access control is delegated using `CODEOWNERS` files
+Each participant in the ecosystem owns their own folder.
+
+Structure:
+
+```
+OpenLineage/spec/registry/
+	/{Name}/           <- custom facet schemas are stored in this folder
+	                   <- and respect the same rules as spec/facets
+		CODEOWNERS     <- Delegate approval to the owners of the name
+		registry.json: <- one file per participant
+			Producer: 
+				Producer root doc URL: https://…
+				Produced facets:
+					{ facets: [ {“URI}”, “{URI}”, … ]}
+			Consumer: 
+				Consumer root doc URL: https://…
+				Consumed facets:
+					{ facets: [ “{URI}”, “{URI}”, … ]}
+		/facets/       <- where custom facet schemas are stored
+                /facets/examples/{FacetName}/{number}.json       <- where facet examples are stored
+```
+Facet examples are  currently in [spec/tests](https://github.com/OpenLineage/OpenLineage/blob/main/spec/tests/ColumnLineageDatasetFacet/1.json)
+
+Examples:
+
+In `OpenLineage/spec/registry/`
+
+```
+airflow/
+	CODEOWNERS
+	registry.json
+	{
+		producer: {
+			root_doc_URL: “https://airflow/doc”
+			produced_facets: [
+				“ol:airflow:AirflowRunFacet.json”,
+				“ol:core:1-0-0/DatasetVersionDatasetFacet.json”,
+				…
+			]
+		}
+	}
+	facets/AirflowRunFacet.json
+```
+```
+core/
+	CODEOWNERS
+	registry.json
+	{
+		producer: {
+			root_doc_URL: "https://openlineage.io/spec/facets/",
+			sample_URL: "https://github.com/OpenLineage/OpenLineage/tree/main/spec/tests/",
+			facets: [
+				"ColumnLineageDatasetFacet.json": {
+					"owner": "core",
+					"spec_versions": [ "1-0-1", "1-0-0" ],
+					"use_cases": [ "lineage", "catalog" ]
```

Comment:
I think we could have generic `tags` rather than `use_cases` - for example, we could have a tag for a dataset facets describing some physical properties of a dataset.

Reply:
Sure, that works - a use case is one category of tags, and since they're not exclusive we can absolutely have a more generic "tag".

Reply:
I think it's a great idea.
How about we create a separate github issue to define tags in the registry?
I'll remove the "use_cases" field from this proposal and @mobuchowski or @Sheeri, you create an issue to describe how we'd add tags to the registry. That way, I can unblock @harels getting the registry started and we give a proper description to the tag feature.